FIAT/raviart_thomas.py

```diff
@@ -76,45 +75,35 @@ def __init__(self, ref_el, degree, variant, quad_deg):
             facet = ref_el.get_facet_element()
             # Facet nodes are \int_F v\cdot n p ds where p \in P_{q-1}
             # degree is q - 1
-            Q = quadrature.make_quadrature(facet, quad_deg)
-            Pq = polynomial_set.ONPolynomialSet(facet, degree)
-            Pq_at_qpts = Pq.tabulate(Q.get_points())[tuple([0]*(sd - 1))]
-            for f in range(len(t[sd - 1])):
-                for i in range(Pq_at_qpts.shape[0]):
-                    phi = Pq_at_qpts[i, :]
-                    nodes.append(functional.IntegralMomentOfScaledNormalEvaluation(ref_el, Q, phi, f))
+            Q = create_quadrature(facet, interpolant_deg + degree - 1)
+            Pq = polynomial_set.ONPolynomialSet(facet, degree - 1)
+            Pq_at_qpts = Pq.tabulate(Q.get_points())[(0,)*(sd - 1)]
+            nodes.extend(functional.IntegralMomentOfScaledNormalEvaluation(ref_el, Q, phi, f)
+                         for f in range(len(t[sd - 1]))
+                         for phi in Pq_at_qpts)
 
             # internal nodes. These are \int_T v \cdot p dx where p \in P_{q-2}^d
-            if degree > 0:
-                Q = quadrature.make_quadrature(ref_el, quad_deg)
-                qpts = Q.get_points()
-                Pkm1 = polynomial_set.ONPolynomialSet(ref_el, degree - 1)
-                zero_index = tuple([0 for i in range(sd)])
-                Pkm1_at_qpts = Pkm1.tabulate(qpts)[zero_index]
-
-                for d in range(sd):
-                    for i in range(Pkm1_at_qpts.shape[0]):
-                        phi_cur = Pkm1_at_qpts[i, :]
-                        l_cur = functional.IntegralMoment(ref_el, Q, phi_cur, (d,), (sd,))
-                        nodes.append(l_cur)
+            if degree > 1:
```

Comment:
Again, not clear what's going with switching choice from 0 to 1?  Maybe 20 years ago I wrote the code numbering from 0 and then it got translated into Norwegian?

Reply:
Correct.